cmd/lncli/main.go

```diff
@@ -72,7 +88,48 @@ func getClientConn(ctx *cli.Context) *grpc.ClientConn {
 	if err != nil {
 		fatal(err)
 	}
-	opts := []grpc.DialOption{grpc.WithTransportCredentials(creds)}
+
+	// We want to prioritize the command line-provided macaroon file first,
+	// then the admin macaroon file, then the read-only macaroon file. This
+	// lets us try to get the highest possible permissions by default when
+	// the command line doesn't specify a macaroon file to use for auth.
+	macPath := cleanAndExpandPath(ctx.GlobalString("macaroonpath"))
+	if !fileExists(macPath) {
```

Comment:
If the command line specifies a macaroon file, but that file doesn't exist due to a typo, could this admin fallback approach unintentionally escalate privilege?